src/infrastructure/implementations/crux-messenger.ts

```diff
@@ -186,9 +186,9 @@ export class CruxNetPubSubClientFactory implements IPubSubClientFactory {
         };
     }
     public getClient = (from: CruxId, keyManager: IKeyManager, to?: CruxId): IPubSubClient => {
-        if (this.bufferPahoClient) { return this.bufferPahoClient; }
+        if (this.bufferPahoClient[from.toString()]) { return this.bufferPahoClient[from.toString()]; }
```

Comment:
can you try to not `from.toString()` so many times in this method. Just do it once and use the string value.

Reply:
done